name: Pytest

on: [push]

jobs:
  build:

    runs-on: ubuntu-20.04

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python 3.8
      uses: actions/setup-python@v1
      with:
        python-version: 3.8
    - name: install gdal
      run: |
        sudo apt-add-repository ppa:ubuntugis/ubuntugis-unstable
        sudo apt-get update
        sudo apt-get install libgdal-dev
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        # pin virtualenv pipenv b/c of problem with use_2to3 in setuptools 58.*
        # pip install --upgrade --no-cache-dir setuptools pip
        pip3 install --no-cache-dir .[test]
    - name: Test with pytest
      run: |
        pytest -vv
